core-bundle/src/Filesystem/Dbafs/Dbafs.php

```diff
@@ -314,11 +314,20 @@ private function doComputeChangeSet(array $dbPaths, array $allDbHashesByPath, ar
         /** @var array<string|int, array<string>> $dirHashesParts */
         $dirHashesParts = [];
 
-        /** @var array<string, int|null> $lastModifiedUpdates */
+        /** @var array<string|int, int|null> $lastModifiedUpdates */
         $lastModifiedUpdates = [];
 
         $isPartialSync = \count($dbPaths) !== \count($allDbHashesByPath);
 
+        // Historically, PHP's ksort() function put numeric indices last, but
+        // this behavior was changed in a bugfix release (see
+        // https://github.com/php/php-src/issues/9296). In order to get stable
+        // results, we use our own implementation, that mimics the old behavior.
+        $ksort = static fn (array &$array) => uksort(
+            $array,
+            static fn ($a, $b) => is_numeric($a) <=> is_numeric($b) ?: $a <=> $b
+        );
+
```

Comment:
We might also use `SORT_NATURAL` here, instead, and introduce a change…

Reply:
:+1: for introducing a change here, but if we’d switch to `SORT_STRING` the numer of affected setups should be pretty low (I think, or at least hope 😅)

Reply:
Do you have a clue what would be closest to the MySQL sorting (`name` column of `tl_files`), at least with the default charset/collation? This is what we should use for the hash part IMHO, because of its use in the legacy code.

After seeing how very crazy inconsistent the default output is, I'm all for a break for the other usage.

Reply:
> Do you have a clue what would be closest to the MySQL sorting

AFAIK `SORT_STRING` should be closest.